File: src/test/java/org/micoli/php/builders/WatcherBuilder.java
package org.micoli.php.builders;

import org.micoli.php.tasks.configuration.Watcher;

public class WatcherBuilder {
    private final Watcher watcher;

    private WatcherBuilder() {
        this.watcher = new Watcher();
    }

    public static WatcherBuilder create() {
        return new WatcherBuilder();
    }

    public WatcherBuilder withTaskId(String taskId) {
        watcher.taskId = taskId;
        return this;
    }

    public WatcherBuilder withDebounce(int debounce) {
        watcher.debounce = debounce;
        return this;
    }

    public WatcherBuilder withNotify(boolean notify) {
        watcher.notify = notify;
        return this;
    }

    public WatcherBuilder withWatches(String[] watches) {
        watcher.watches = watches;
        return this;
    }

    public Watcher build() {
        return watcher;
    }
}

File: src/test/kotlin/org/micoli/php/builders/WatcherBuilder.kt
package org.micoli.php.builders

import org.micoli.php.tasks.configuration.Watcher

class WatcherBuilder private constructor() {
    private val watcher: Watcher = Watcher()

    fun withTaskId(taskId: String): WatcherBuilder {
        watcher.taskId = taskId
        return this
    }

    fun withDebounce(debounce: Int): WatcherBuilder {
        watcher.debounce = debounce
        return this
    }

    fun withNotify(notify: Boolean): WatcherBuilder {
        watcher.notify = notify
        return this
    }

    fun withWatches(watches: Array<String>): WatcherBuilder {
        watcher.watches = watches
        return this
    }

    fun build(): Watcher {
        return watcher
    }

    companion object {
        fun create(): WatcherBuilder {
            return WatcherBuilder()
        }
    }
}
